build.sbt

```diff
@@ -77,17 +82,61 @@ lazy val commonSettings = Seq(
   }
 )
 
-val release = Seq(
-  bintrayOrganization := Some("azavea"),
-  bintrayRepository := "maven",
-  bintrayVcsUrl := Some("https://github.com/geotrellis/vectorpipe.git"),
+lazy val noPublishSettings = Seq(
+  publish := {},
+  publishLocal := {},
+  publishArtifact := false
+)
+
+lazy val publishSettings = Seq(
+  organization := "com.azavea.geotrellis",
+  organizationName := "GeoTrellis",
+  organizationHomepage := Some(new URL("https://geotrellis.io/")),
+  description := "Import OSM data and output to VectorTiles with GeoTrellis.",
+  publishArtifact in Test := false
+) ++ sonatypeSettings ++ credentialSettings
+
+lazy val sonatypeSettings = Seq(
   publishMavenStyle := true,
-  publishArtifact in Test := false,
-  pomIncludeRepository := { _ => false },
-  licenses += ("Apache-2.0", url("http://apache.org/licenses/LICENSE-2.0")),
-  homepage := Some(url("https://geotrellis.github.io/vectorpipe/"))
+
+  sonatypeProfileName := "com.azavea",
+  sonatypeProjectHosting := Some(GitHubHosting(user="geotrellis", repository="vectorpipe", email="systems@azavea.com")),
+  developers := List(
+    Developer(id = "jpolchlo", name = "Justin Polchlopek", email = "jpolchlopek@azavea.com", url = url("https://github.com/jpolchlo")),
+    Developer(id = "mojodna", name = "Seth Fitzsimmons", email = "seth@mojodna.net", url = url("https://github.com/mojodna"))
+  ),
+  licenses := Seq("Apache-2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+
+  publishTo := sonatypePublishTo.value
+)
+
+lazy val credentialSettings = Seq(
+  credentials += Credentials(
+    "GnuPG Key ID",
+    "gpg",
+    System.getenv().get("GPG_KEY_ID"),
+    "ignored"
+  ),
```

Comment:
@hectcastro Kind of late to the party here, but what is the mechanism that prevents the gpg from asking for the passphrase during the publish? Is this somehow side effect of `gpg --batch` command in `cibuild` ?

Reply:
Possibly unrelated, but `publishLocal` gives
```
[warn] GPG reported an error. Artifacts won't be signed.
```
Also, there is an NPE through log4j raised just before that message pops up.

FFT?

Reply:
Yup, this is related to the line you saw in the CircleCI YAML.

The way we get around GPG asking for a passphrase is super simple: we simply do not password-protect the signing key. This is OK, because the signing key is a subkey that is only used in CI and is very easy to invalidate.

Better docs on this workflow will be dropping soon in the Operations repo.

See: https://github.com/jodersky/sbt-gpg#securely-publishing-in-travis-and-other-ci

Reply:
@jpolchlo It was pulling in an env var that was not defined, thus producing `null` value, I added a commit to flatten credentials from vars that don't exist.